doonut-app-external-api/src/main/kotlin/com/doonutmate/member/service/MemberAppService.kt

```diff
@@ -1,16 +1,31 @@
 package com.doonutmate.member.service
 
+import com.doonutmate.doonut.member.model.OauthType
 import com.doonutmate.doonut.member.service.MemberBusinessService
-import org.springframework.beans.factory.annotation.Autowired
+import com.doonutmate.member.controller.dto.DeleteRequest
+import com.doonutmate.member.service.strategy.AppleMemberDelete
+import com.doonutmate.member.service.strategy.KakaoMemeberDelete
 import org.springframework.stereotype.Service
 
 @Service
 class MemberAppService(
-    @Autowired var memberBusinessService: MemberBusinessService,
+    private val appleMemberDelete: AppleMemberDelete,
+    private val kakaoMemeberDelete: KakaoMemeberDelete,
+    private val memberBusinessService: MemberBusinessService,
 ) {
-    fun delete(memberId: Long) {
-        memberBusinessService.delete(memberId)
+    fun delete(req: DeleteRequest) {
+        when (req.oauthType) {
+            OauthType.KAKAO -> {
+                kakaoMemeberDelete.delete(req)
+            }
```

Comment:
스프링 DI와 다형성을 이용하면 분기문을 없앨 수 있습니다!

```suggestion
    private val deleteStrategies: List<MemberDeleteStrategy>,
    private val memberBusinessService: MemberBusinessService,
) {
    fun delete(req: DeleteRequest) {
        deleteStrategies.first { it.supports(req.oauthType) }
            .delete(req)
    }
```